src/realm/mixed.cpp

```diff
@@ -756,6 +756,54 @@ StringData Mixed::get_index_data(std::array<char, 16>& buffer) const noexcept
     return {};
 }
 
+std::string Mixed::to_string(size_t max_size) const noexcept
+{
+    std::ostringstream ostr;
+    if (is_type(type_String)) {
+        std::string ret = "\"";
+        if (string_val.size() <= max_size) {
+            ret += std::string(string_val);
+        }
+        else {
+            ret += std::string(StringData(string_val.data(), max_size)) + " ...";
+        }
+        ret += "\"";
+        return ret;
+    }
+    else if (is_type(type_Binary)) {
+
+        static char hex_chars[] = "0123456789ABCDEF";
+        auto out_hex = [&ostr](char c) {
+            ostr << hex_chars[c >> 4];
+            ostr << hex_chars[c & 0xF];
+            ostr << ' ';
+        };
+
+        auto sz = binary_val.size();
+        bool capped = false;
+        if (3 * sz > max_size) {
```

Comment:
Can you put this magic number in some variable, just be easier to understand what 3 means in this context?

Reply:
Done